### DISCUSSION TITLE
Windows(OS)

### DISCUSSION THREAD

**Discussion by Xyndra** [hyprwm/Hyprland#4333] [Q&A] [answered]
Is there any way to get this working on Windows or is that a stupid question?

**Comment by vaxerski** [comment]
it is a stupid question. No.

**Reply by VoiderYT** [reply]
emotional damage

**Reply by MabaRod** [reply]
There exist some people (like me) that love the Hyprland experience, and we would love to see it on windows. Its not a stupid question. There are no stupid questions, just stupid people that don't ask. Have more respect for people.  :)

In my case i switch back to windows because of work, yet i love the KDE and the Hyprland experience, i believe it should be nice to get compositors on windows.

**Reply by yavko** [reply]
> There exist some people (like me) that love the Hyprland experience, and we would love to see it on windows. Its not a stupid question. There are no stupid questions, just stupid people that don't ask. Have more respect for people.  :)
> 
> In my case i switch back to windows because of work, yet i love the KDE and the Hyprland experience, i believe it should be nice to get compositors on windows.

Well it won't happen, and it doesn't make sense to.

**Reply by MabaRod** [reply]
In not trying to start a fight, but why we could not get a Hyprland style wm on windows? Or at least some personalization options, I'm asking for opinions from more experienced developers.

**Reply by yavko** [reply]
I explain why in my other comments in this discussion.

**Comment by yavko** [comment, marked as answer]
Hyprland is Wayland compositor for unix-like systems, Wayland being the protocol that applications (clients) use to display stuff. However, windows already has its own compositor (and you can easily modify anything in windows) which wouldn't make Hyprland a viable thing to run on windows. Also as I previously mentioned Hyprland is for unix-like systems, which operate fundamentally differently than windows, which would require a very difficult port for something totally impractical. And lastly applications built for windows wouldn't work with this Hyprland port as windows applications use an API that is completely different than the Wayland protocol.  I could be wrong on some things, but I hope this explains why there is no way to run it on my windows.

**Reply by yavko** [reply]
Also the Wayland protocol is incompatible with windows due to windows's UDS implementation not supporting ancillary data (required for wayland), probably due to handling file descriptors in a completely different way.

**Comment by abhaythakur71181** [comment]
lol .
its just impossible

**Reply by maxwick456** [reply]
wsl

**Reply by dottie-softs** [reply]
> wsl

bro but wsl is just a terminal without display manager lmao

**Reply by BryanDollery** [reply]
> > wsl
> 
> bro but wsl is just a terminal without display manager lmao

That isn't true. WSL is a full linux distro, and it has had a window manager for years now. It still can't run hyprland, though sway works.

**Reply by TRPB** [reply]
Do we know what makes sway work? It's a wayland compositor like Hyprland, what needs to be changed in Hyprland to work in the same way? See https://github.com/jordankoehn/sway-wsl2

**Reply by yavko** [reply]
> Do we know what makes sway work? It's a wayland compositor like Hyprland, what needs to be changed in Hyprland to work in the same way? See https://github.com/jordankoehn/sway-wsl2 

Afaik it's because Hyprland doesn't support software rendering.

**Comment by BryanDollery** [comment]
Impossible is a strong word. Windows comes with WSL (which is basically a well integrated VM running linux), and WSL comes with X11. I can run gimp on windows! It's great. But, right now, afaik, it's still not possible right now to run hyprland on windows because the whole X11 thing is (somehow) completely novel to MS and they've not made it quite as flexible as it needs to be (yet).

However, with some changes by MS, it may become possible to run hyprland on windows.

**Reply by yavko** [reply]
That's not the case. Wsl2 includes wslg, a Wayland server with xWayland support, which renders linux apps as windows apps, however this has nothing to do with X11. The real reason Hyprland doesn't work on wsl, is because wsl does not provide DRM drivers.

**Reply by yavko** [reply]
Also the reason DRM drivers are required, is because Hyprland requires hardware accelerated rendering, and requires DRM for that. Sway doesn't so you can run it however even if you do manage to get it working it is not the same. You'll only get it running in windowed mode, and you'll be using wsl for everything anyways, so what's the point of just not installing linux, it's kind of useless.

**Reply by mattmcadoo** [reply]
> you'll be using wsl for everything anyways, so what's the point of just not installing linux, it's kind of useless.

For those that have use a company/corporate system with an approved OS, WSL is the closest means to getting a real Linux distro.  (Even if Mac is offered, WSL is a lot times better to use.)

**Reply by yavko** [reply]
> > you'll be using wsl for everything anyways, so what's the point of just not installing linux, it's kind of useless.
> 
> For those that have use a company/corporate system with an approved OS, WSL is the closest means to getting a real Linux distro.  (Even if Mac is offered, WSL is a lot times better to use.)

That doesnt change the fact that hyprland still wont run.

**Comment by cuxdi** [comment]
>  wsl does not provide DRM drivers.

this may be  a dumb question, but cant you just change some code and port hyprland to wsl?

**Comment by Schievel1** [comment]
Well it's obviously not "running on windows", but I got sway running in WSL2.

**Reply by iceking-1912** [reply]
how??
is hyprland also possible

**Reply by keksc** [reply]
bro read previous responses oh my days https://github.com/hyprwm/Hyprland/discussions/4333#discussioncomment-9306809

**Comment by Jervi-sir** [comment]
m a bit late, maybe check this comment i found on reddit

> Install YASB https://github.com/amnweb/yasb
> Install Komorebi https://github.com/LGUG2Z/komorebi
> 
> https://www.reddit.com/r/desktops/comments/1gbbh91/yasb_komorebi/

> I also found this repo: https://github.com/glzr-io/glazewm
